### DISCUSSION TITLE
Choosing a Specific Variable for Simulation in OpenOil

### DISCUSSION THREAD

**Discussion by flaphio** [OpenDrift/opendrift#1198] [Q&A]
Hi all,

I've been working on simulating oil spills using OpenOil with Copernicus Marine data on the Brazilian Equatorial Margin. Currently, the simulation automatically selects variables, but I want to take control and manually choose the variables for the simulation. Here's the structure of the Mercator data I'm working with:

Reader: E:/OpenDrift/4. Reader_MEB/mercator/GAF_2023_01.nc
Projection:
  +proj=latlong
Coverage: [degrees]
  xmin: -55.000000   xmax: -30.000000   step: 0.0833321   numx: 301
  ymin: -6.000000   ymax: 6.000000   step: 0.0833335   numy: 144
  Corners (lon, lat):
    (-55.00,   6.00)  (-30.00,   6.00)
    (-55.00,  -6.00)  (-30.00,  -6.00)
Vertical levels [m]:
  [-0.494025]
Available time range:
  start: 2023-01-01 00:30:00   end: 2023-01-31 23:30:00   step: 1:00:00
    744 times (0 missing)
Variables:
  surface_sea_water_x_velocity
  sea_surface_wave_stokes_drift_y_velocity
  sea_surface_wave_stokes_drift_x_velocity
  northward_sea_water_velocity
  surface_sea_water_y_velocity_due_to_tide
  eastward_sea_water_velocity
  surface_sea_water_x_velocity_due_to_tide
  surface_sea_water_y_velocity
  y_sea_water_velocity
  x_sea_water_velocity
  sea_water_speed - derived from ['x_sea_water_velocity', 'y_sea_water_velocity']
  sea_surface_wave_stokes_drift_speed - derived from ['sea_surface_wave_stokes_drift_x_velocity', 'sea_surface_wave_stokes_drift_y_velocity']
===========================
 0:00:45.0  total
 0:00:00.9  preparing
 0:00:43.9  reading
 0:00:06.8  interpolation
 0:00:00.2  interpolation_time
         0  masking
         
         
Currently, the simulation automatically selects variables, but I want to manually configure it. How can I achieve this? Your guidance and suggestions would be greatly appreciated.

Thank you!

**Comment by knutfrode** [comment]
You can create a reader like normal, and then specify which variables should be used when adding it to the simulation, e.g.
`o.add_reader(reader_atmosphere, variables=['x_wind', 'y_wind'])`

**Reply by flaphio** [reply]
I followed your recommendation, but unfortunately, I didn't achieve the desired results. My current challenge involves attempting to access a specific variable from the dataset. Despite my efforts, I'm encountering difficulties in obtaining the expected outcome. Any further guidance or clarification would be greatly appreciated.

The structure of the data is as follows:"

Reader: /home/flavioborges/Documentos/Opendrift/4. Reader_MEB/mercator_data/gaf_wave_202101.nc
Projection: 
  +proj=latlong
Coverage: [degrees]
  xmin: -55.000000   xmax: -30.000000   step: 0.0833321   numx: 301
  ymin: -6.000000   ymax: 10.000000   step: 0.0833335   numy: 192
  Corners (lon, lat):
    (-55.00,  10.00)  (-30.00,  10.00)
    (-55.00,  -6.00)  (-30.00,  -6.00)
Vertical levels [m]: 
  [-0.494025]
Available time range:
  start: 2021-01-01 00:30:00   end: 2021-01-31 22:30:00   step: 1:00:00
    743 times (0 missing)
Variables:
  sea_surface_wave_stokes_drift_y_velocity
  sea_surface_wave_stokes_drift_x_velocity
  northward_sea_water_velocity
  eastward_sea_water_velocity
  y_sea_water_velocity
  x_sea_water_velocity
  sea_water_speed - derived from ['x_sea_water_velocity', 'y_sea_water_velocity']
  sea_surface_wave_stokes_drift_speed - derived from ['sea_surface_wave_stokes_drift_x_velocity', 'sea_surface_wave_stokes_drift_y_velocity']
===========================


I am attempting to access ['sea_surface_wave_stokes_drift_x_velocity', 'sea_surface_wave_stokes_drift_y_velocity'], but I encountered the following issue:

ValueError: Readers must be added for the following required variables: ['x_sea_water_velocity', 'y_sea_water_velocity']

**Comment by knutfrode** [comment]
It is not totally clear what you are trying to achieve.
A given model needs some certain variables (`o.required_variables`) to run, e.g. wind (`x_wind`, `y_wind`) and current (`x_sea_water_velocity`, `y_sea_water_velocity`)
These must be provided by some readers, or you can use constant values: `set_config('environment:constant:<variable>, <value>)`
It seems that you are trying to start a simulation without adding a reader providing currents?
